### PR TITLE
[rs] 08: Restructure the entire project.

### PR DESCRIPTION
**Pull request by Rhuagh**
This contains all the changes from #424 #425 #426 #427, and additional formatting fixes and cleanup.

### CODE REVIEW

**Comment by Xaeroxe** [comment]
The big one!  Once 24 hours have elapsed for the other pull requests if we haven't gotten any change requests I'll r+ this.

File: src/util/time.rs [hunk 1]
```diff
@@ -1,13 +0,0 @@
-//! Util time conversion functions
-
-use std::time::Duration;
-
-///Converts a Duration to the time in seconds.
-pub fn duration_to_secs(duration: Duration) -> f32 {
```
Comment (jojolepro) [inline src/util/time.rs]: Bring it back? :D

Comment (Rhuagh) [reply src/util/time.rs]: `core::timing::*`

Comment (Rhuagh) [reply src/util/time.rs]: Hmm, where did that to :(

**Comment by Xaeroxe** [comment]
Some of the fix commits from the other PRs are missing, @Rhuagh has already indicated they plan to fix this.

**Comment by Rhuagh** [comment]
@jojolepro r?

**Comment by Xaeroxe** [comment]
Please note: This is a roll-up PR and thus not subject to our normal merging rules.

**Pull request merged**